bin/live/pycbc_live_combine_single_fits

```diff
@@ -49,23 +53,44 @@ if args.conservative_percentile < 50 or \
                  "conservative.")
 
 colours = ['r', (1.0, 0.6, 0), 'y', 'g', 'c', 'b', 'm',
-           (0.8, 0.25, 0), (0.25, 0.8, 0)]
+           (0.25,0.9,0)]
+# If the above colours are exhausted, revert to the default colour cycle
+colours += plt.rcParams['axes.prop_cycle'].by_key()['color']
 
 counts_all = {ifo:[] for ifo in args.ifos}
 alphas_all = {ifo:[] for ifo in args.ifos}
 analysis_dates = []
 
-files = args.trfits_files
+with h5py.File(args.trfits_files[0], 'r') as fit_f0:
+    # Store some attributes so we can check that all files are
+    # comparable
 
-fit_f0 = h5py.File(files[0], 'r')
+    # Keep the upper and lower bins
+    bl = fit_f0['bins_lower'][:]
+    bu = fit_f0['bins_upper'][:]
 
-bl = fit_f0['bins_lower'][:]
-bu = fit_f0['bins_upper'][:]
+    sngl_rank = fit_f0.attrs['sngl_ranking']
+    fit_thresh = fit_f0.attrs['fit_threshold']
+    fit_func = fit_f0.attrs['fit_function']
 
 live_times = {ifo : [] for ifo in args.ifos}
 
-for f in files:
+logging.info("Checking through %d files", len(args.trfits_files))
+
+epoch = datetime.date(2000,1,1)
+for f in args.trfits_files:
+
+    # Get the date information, there are a couple of ways to do this:
```

Comment:
This comment seems more useful just before line 94.

Reply:
Also, I would add in the comment a short reminder of why we need to do all that date logic - why can we not simply look at the GPS times of the triggers?

Reply:
Wait, that's a much better way to do this. How did I not think of that before.
I suppose we may not always decide to output the time of the triggers which we used in the fitting from the other code, but we probably will do, especially now the plotting has been split off